### PR TITLE
Print errors when the Databricks run fails to start

### PR DESCRIPTION
**Pull request by andrewmchen**
If the cluster spec is wrong, you'll usually get an error looking like
```
....
  File "/Users/andrew/mlflow/env/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/andrew/mlflow/env/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/andrew/mlflow/mlflow/cli.py", line 110, in run
    run_id=run_id,
  File "/Users/andrew/mlflow/mlflow/projects/__init__.py", line 121, in run
    storage_dir=storage_dir, block=block, run_id=run_id)
  File "/Users/andrew/mlflow/mlflow/projects/__init__.py", line 54, in _run
    experiment_id=exp_id, cluster_spec=cluster_spec)
  File "/Users/andrew/mlflow/mlflow/projects/databricks.py", line 261, in run_databricks
    db_run_id = _run_shell_command_job(uri, command, env_vars, cluster_spec)
  File "/Users/andrew/mlflow/mlflow/projects/databricks.py", line 192, in _run_shell_command_job
    databricks_run_id = run_submit_res["run_id"]
KeyError: 'run_id'
```
Let's catch the `KeyError` and throw something nicer.

### CODE REVIEW

File: mlflow/projects/databricks.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def databricks_api_request(self, endpoint, method, **kwargs):
         return json.loads(response.text)
 
     def _jobs_runs_submit(self, json):
+        print 'here'
```
Comment (smurching) [inline mlflow/projects/databricks.py]: Nit: remove this

**Review by smurching — COMMENTED** [review]
Generally looks good, thanks for the catch! That being said I wonder if we could handle failed requests in the `databricks_api_request` method (https://github.com/mlflow/mlflow/blob/HEAD/mlflow/projects/databricks.py#L50), since there are other places in the code where we currently assume a successful request (e.g. here https://github.com/mlflow/mlflow/blob/HEAD/mlflow/projects/databricks.py#L309) when hitting the Databricks API. One way could be to check for an `error_code` field, although not sure if we can assume the existence of such a field.

**Comment by andrewmchen** [comment]
Yeah that's a good point. The right way is probably to check the HTTP status code of the response.

**Review by smurching — COMMENTED** [review]
This LGTM pending the one comment from offline discussion about making DatabricksJobRunner._upload_to_dbfs also use the _databricks_rest_api interface / checking that the upload requests succeeds (has status 200). Thanks @andrewmchen!

**Review by smurching — APPROVED** [review]
LGTM thanks @andrewmchen!

**Pull request merged**